python/foglamp/services/core/api/notification.py

```diff
@@ -0,0 +1,517 @@
+                "notification_type": notification_config['notification_type']['value'],
+                "enable": notification_config['enable']['value'],
+            }
+            notifications.append(notification)
+    except Exception as ex:
+        raise web.HTTPInternalServerError(reason=ex)
+    else:
+        return web.json_response({'notifications': notifications})
+
+
+async def post_notification(request):
+    """
+    Create a new notification to run a specific plugin
+
+    :Example:
+             curl -X POST http://localhost:8081/foglamp/notification -d '{"name": "Test Notification", "description":"Test Notification", "rule": "threshold", "channel": "email", "notification_type": "one shot", "enabled": false}'
+             curl -X POST http://localhost:8081/foglamp/notification -d '{"name": "Test Notification", "description":"Test Notification", "rule": "threshold", "channel": "email", "notification_type": "one shot", "enabled": false, "ruleConfig": {}, "deliveryConfig": {}}'
+    """
+    try:
+        notification_service = ServiceRegistry.get(s_type=ServiceRecord.Type.Notification.name)
+        _address, _port = notification_service[0]._address, notification_service[0]._management_port
+    except service_registry_exceptions.DoesNotExist:
+        return {'result': "No Notification service available."}
+
+    try:
+        data = await request.json()
+        if not isinstance(data, dict):
+            raise ValueError('Data payload must be a valid JSON')
+
+        name = data.get('name', None)
+        description = data.get('description', None)
+        rule = data.get('rule', None)
+        channel = data.get('channel', None)
+        notification_type = data.get('notification_type', None)
+        enabled = data.get('enabled', None)
+        rule_config = data.get('ruleConfig', {})
+        delivery_config = data.get('deliveryConfig', {})
+
+        if name is None:
+            raise web.HTTPBadRequest(reason='Missing name property in payload.')
+        if description is None:
+            raise web.HTTPBadRequest(reason='Missing description property in payload.')
+        if rule is None:
+            raise web.HTTPBadRequest(reason='Missing rule property in payload.')
+        if channel is None:
+            raise web.HTTPBadRequest(reason='Missing channel property in payload.')
+        if notification_type is None:
+            raise web.HTTPBadRequest(reason='Missing notification_type property in payload.')
+
+        if utils.check_reserved(name) is False:
+            raise web.HTTPBadRequest(reason='Invalid name property in payload.')
+        if utils.check_reserved(rule) is False:
+            raise web.HTTPBadRequest(reason='Invalid rule property in payload.')
+        if utils.check_reserved(channel) is False:
+            raise web.HTTPBadRequest(reason='Invalid channel property in payload.')
+        if notification_type not in ["one shot", "retriggered", "toggled"]:
```

Comment:
It should declare as enum
```
class NotificationType(IntEnum):
    """ Enumeration for Notification type """
    oneshot = 0
    retriggered = 1
    toggled = 2
```

Reply:
notification_type is part of configuration options, hence cannot be declared separately via a class.

Reply:
let's move this as a const on TOP of this module `NOTIFICATIONS_TYPE = ["one shot", "retriggered", "toggled"]`